### PR TITLE
Binder ordering no longer matters

### LINKED ISSUE #18: Ignore ordering in mutually recursive blocks
Currently, this example will fail.

```
inf = go0
  where
    go0 = 'a' : go1
    go1 = 'b' : go2
    go2 = 'c' : go0
inf' = go0
  where
    go1 = 'b' : go2
    go0 = 'a' : go1
    go2 = 'c' : go0

 inspect $ 'inf === 'inf'
```

but it doesn't seem like it should.

### PR DESCRIPTION
**Pull request by LightAndLight**
This fixes #18

### CODE REVIEW

File: Test/Inspection/Core.hs [hunk 1]
```diff
@@ -91,11 +91,16 @@ type VarPairSet = S.Set VarPair
...
 -- It also breaks if there is shadowing.
-eqSlice :: Bool {- ^ ignore types -} -> Slice -> Slice -> Bool
-eqSlice _ slice1 slice2 | null slice1 || null slice2 = null slice1 == null slice2
+eqTerm
+  :: Bool {- ^ ignore types -}
+  -> (Var, Slice) -- ^ First variable, and the bindings it gives rise to
+  -> (Var, Slice) -- ^ Second variable, and the bindings it gives rise to
+  -> Bool
+eqTerm _ (_, slice1) (_, slice2)
+  | null slice1 || null slice2 = null slice1 == null slice2
```
Comment (nomeata) [inline Test/Inspection/Core.hs]: Is this case still needed now that there is no longer a call to `last`?

**Comment by LightAndLight** [comment]
I just realised I didn't fix the letrec case

**Comment by nomeata** [comment]
Ok, will hold off merging for now.

**Comment by LightAndLight** [comment]
I've started trying to fix it, but I'm a little stuck. I thought this would work https://github.com/LightAndLight/inspection-testing/blob/binder-ordering-wip/Test/Inspection/Core.hs#L94

but it fails two tests. I need to go do other things for a while, so I'm leaving it here in case you have any ideas.

**Comment by nomeata** [comment]
@LightAndLight Sorry for dropping the ball on this. Is this still something you are working on?

**Comment by LightAndLight** [comment]
No worries. I have't touched it since last year

**Comment by isovector** [comment]
I just ran into this bug when trying to test a core plugin, something I'm particularly keen to make sure I got right. Since I can't control the order of the binding groups, this is a pretty big blocker for me! 

The PR seems pretty dead --- do either of you (@nomeata , @LightAndLight ) want a hand with it?

**Comment by nomeata** [comment]
Fixed via https://github.com/nomeata/inspection-testing/commit/30d76bfe521bbdc98b4a60a25461be784ba4b17d instead, please let me know if that does not address your needs, and thanks for your contribution.

**Pull request closed without merging**